packages/stylelint-plugin/src/rules/no-invalid-use-of-tokens.ts

```diff
@@ -0,0 +1,354 @@
+import { AtRule, Declaration, Root } from "postcss"
+import postcssValueParser from "postcss-value-parser"
+import { deprecatedTokenReplacements } from "../../deprecatedTokens"
+import {
+  cantReplaceDeprecatedTokenInAtRuleMessage,
+  cssVariableUsedWithinUnsupportedFunction,
+  deprecatedTokenInVariableMessage,
+  deprecatedTokenUsageMessage,
+  deprecatedTokenUsageWithoutReplacementMessage,
+  genericContainsDeprecatedKaizenTokenMessage,
+  invalidEquationContainingKaizenTokenMessage,
+  replacementCssVariableUsedWithinUnsupportedFunction,
+} from "../messages"
+import { Options, RuleDefinition } from "../types"
+import { cssStandardFunctions } from "../util/cssStandardFunctions"
+import { kaizenTokensByName } from "../util/kaizenTokens"
+import {
+  isVariable,
+  replaceTokenInVariable,
+  stringifyVariable,
+} from "../util/variableUtils"
+import { walkVariablesOnValue } from "../util/walkers"
+import { fixAlphaModificationFunctions } from "./lib/tokensInFunctions"
+import { containsEquationThatDoesntWorkWithCSSVariables } from "./no-invalid-use-of-var-tokens-in-equations"
+
+const deprecatedKaizenTokenPattern = new RegExp(
+  Array.from(deprecatedTokenReplacements.keys()).join("|")
+)
+export function stringContainsDeprecatedKaizenToken(value: string) {
```

Comment:
Check if this needs to be exported
```suggestion
function stringContainsDeprecatedKaizenToken(value: string) {
```